### PR TITLE
InferredView fixes and enhancements

### PR DESCRIPTION
**Pull request by imanjra**
## What changes are proposed in this pull request?

Fix exceptions for unknown value type and ensure defaults are propagated correctly

## How is this patch tested? If it is not, please explain why.

Using an operator with InferredView rendering a deeply nested object

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: app/packages/core/src/plugins/SchemaIO/utils/generate-schema.ts [hunk 1]
```diff
@@ -16,10 +16,9 @@ export function generateSchema(value, options?) {
           label,
           component: "TableView",
           columns: getTableColumns(value),
-          readOnly: true,
         },
       };
-    } else if (isPrimitiveType(dominantType) && readOnly) {
+    } else if (isPrimitiveType(dominantType || "") && readOnly) {
```
Comment (ritch) [inline app/packages/core/src/plugins/SchemaIO/utils/generate-schema.ts:21]: This is probably the crux of the fix, but I find it very hard to follow. Its odd that `getDominantType()` returns null(ish) values? A variable name or comment indicating that the `||` is due to it being an empty array (isn't that the only case where this happens?

Also I realize we're using existing utils here, but I'd rather have metadata be in object form than functions, the data is already there so its a bit odd to call functions to get at it:

```typescript
const value = "foo";
const typeInfo = getTypeInfoForValue(value);
// typeInfo.isArray => false
// typeInfo.dominantType => null (rename to typeInfo.arrayItemDominantType)
// typeInfo.isPrimitive => true
// typeInfo.value => "foo"
// typeInfo.isNullish => false
// typeInfo.type => string
// etc
```

Both of these a minor comments, but thought they might be useful feedback.

Comment (imanjra) [reply app/packages/core/src/plugins/SchemaIO/utils/generate-schema.ts:21]: Yah, that's right. dominantType will be unknown for an empty array. Planning to use something like one of view in future for unknown type but it can potentially be a nested object.

Not too sure about the second half of the comment. The `generate-schema` (I should probably rename to `inferSchemaUsingValue(value)` is only used for `InferredView` to generate a `SchemaIO` compatible schema based only on a value.
